mesonbuild/compilers/c.py

```diff
@@ -173,6 +173,12 @@ def get_allow_undefined_link_args(self):
         elif self.id == 'msvc':
             # link.exe
             return ['/FORCE:UNRESOLVED']
+        elif self.id == 'intel':
```

Comment:
why not add this to the `if self.id in ('clang', 'gcc'):` block?

Reply:
I'm also working on support for ICC on windows, which behaves like MSVC, so there isn't a good way using ID to make that happen. Really though we should just break this method up and send the GCC/Clang/ICC windows bits to the GCCLikeCompiler class and the MSVC specific information into the MSVC class. I'll add a patch to do that.